src/drivers/legacy-driver.ts

```diff
@@ -94,9 +111,24 @@ export class LegacyCMakeDriver extends CMakeDriver {
     });
   }
 
-  static async create(cmake: CMakeExecutable, config: ConfigurationReader, kit: Kit|null, workspaceFolder: string | null, preconditionHandler: CMakePreconditionProblemSolver, preferredGenerators: CMakeGenerator[]): Promise<LegacyCMakeDriver> {
+  static async create(cmake: CMakeExecutable,
+                      config: ConfigurationReader,
+                      useCMakePresets: boolean,
+                      kit: Kit|null,
+                      configurePreset: ConfigurePreset | null,
```

Comment:
Why does the legacy driver need these arguments in the constructor?  I didn't think they would be used.  Or do we try to invoke cmake with the values in the presets?

Reply:
They are used the same way in all drivers. `create()` calls `_baseInit()`, which sets the initial value for all presets. And then they are used in driver.ts (but may not be used directly in *-driver.ts). The reason we need them on initialization is that drivers may not exist at the same time as CMakeTools instances.